test/inductor/test_flex_attention.py

```diff
@@ -1209,46 +1269,73 @@ def test_captured_score_mod_aot_eager_gradcheck(
-            (inputs[1].grad, inputs_flex[1].grad, inputs_golden[1].grad),
-            (inputs[2].grad, inputs_flex[2].grad, inputs_golden[2].grad),
+        block_mask = create_block_mask(causal, 1, 1, 2048, 2048)
+        no_sparse_flex = functools.partial(flex_attention, score_mod=causal)
+        score_mod_sparse_flex = functools.partial(
+            flex_attention,
+            score_mod=causal,
+            block_mask=create_block_mask(causal, 1, 1, 2048, 2048),
+        )
+        mask_mod_sparse_flex = functools.partial(
+            flex_attention, block_mask=create_block_mask(causal_mask, 1, 1, 2048, 2048)
+        )
+        for attention_call in [
+            no_sparse_flex,
+            score_mod_sparse_flex,
+            mask_mod_sparse_flex,
         ]:
-            ref_error = rmse(ref, golden)
-            flex_error = rmse(flex, golden)
-            # Note: This has been carefully tested that FlexAttention is within
-            # 10% of the average error of SDPA! Do not bump this tolerance
-            # unless you are absolutely sure you are not worsening the accuracy
-            # of FlexAttention!
-            self.assertTrue(ref_error < flex_error * 1.1)
+            inputs = [
+                torch.randn(
+                    2,
+                    2,
+                    2048,
+                    64,
+                    device="cuda",
+                    dtype=torch.float16,
+                    requires_grad=True,
+                )
+                for _ in range(3)
+            ]
+            gradOut = torch.randn(2, 2, 2048, 64, device="cuda", dtype=torch.float16)
+            out_ref = torch.nn.functional.scaled_dot_product_attention(
+                *inputs, is_causal=True
+            )
+            out_ref.backward(gradOut)
+
+            inputs_flex = [i.detach().clone().requires_grad_(True) for i in inputs]
+            out_flex = torch.compile(attention_call)(*inputs_flex)
+            out_flex.backward(gradOut)
+            inputs_golden = [
+                i.detach().clone().to(dtype=torch.float64).requires_grad_(True)
+                for i in inputs
+            ]
+            out_golden = torch.nn.functional.scaled_dot_product_attention(
+                *inputs_golden, is_causal=True
+            )
+            out_golden.backward(gradOut.to(dtype=torch.float64))
+
+            for ref, flex, golden in [
+                (out_ref, out_flex, out_golden),
+                (inputs[0].grad, inputs_flex[0].grad, inputs_golden[0].grad),
+                (inputs[1].grad, inputs_flex[1].grad, inputs_golden[1].grad),
+                (inputs[2].grad, inputs_flex[2].grad, inputs_golden[2].grad),
+            ]:
+                ref_error = rmse(ref, golden)
+                flex_error = rmse(flex, golden)
+                # Note: This has been carefully tested that FlexAttention is within
+                # 20% of the average error of SDPA! Do not bump this tolerance
```

Comment:
lol

Reply:
What if we make torch SPDA even faster though... :P